index.js

```diff
@@ -1,4 +1,83 @@
 //  in this file create an express application - use the middle-ware built into express
-//  to serve up static files from the public directory (index.html and client.js - you 
+//  to serve up static files from the public directory (index.html and client.js - you
 //  can also serve up css files from the public directory this way if you want)
-//  you need to support a '/trucks' endpoint, and a dynamic route for '/trucks/:name'
\ No newline at end of file
+//  you need to support a "/trucks" endpoint, and a dynamic route for "/trucks/:name"
+
+var express = require("express");
+var trucks = require("./trucks");
+var app = express();
+
+var serveStatic = express.static("public");
+app.use(serveStatic);
+ 
+app.get("/trucks", function(request, response) {
+    var truckStr = "";
+    var truckArray = trucks.getTrucks();
+    for (var i = 0; i < truckArray.length; i++) {
+        truckStr += truckArray[i].name + "<br>";
+    }
+
+    response.send(truckStr);
```

Comment:
Rather than send back a string, just return the raw data. An api will typically provide more data, not less, so the caller/consumer can make a decision about what parts they want to use. That said, it's fine for this assignment.